python/ql/src/Security/CWE-352/CSRFProtectionDisabled.ql

```diff
@@ -0,0 +1,23 @@
+/**
+ * @name CSRF protection weakened or disabled
+ * @description Disabling or weakening CSRF protection may make the application
+ *              vulnerable to a Cross-Site Request Forgery (CSRF) attack.
+ * @kind problem
+ * @problem.severity warning
+ * @security-severity 8.8
+ * @precision high
+ * @id py/csrf-protection-disabled
+ * @tags security
+ *       external/cwe/cwe-352
+ */
+
+import python
+import semmle.python.Concepts
+
+from CsrfProtectionSetting s
+where
+  s.getVerificationSetting() = false and
+  not exists(CsrfLocalProtection p) and
+  // rule out test code as this is a common place to turn off CSRF protection
+  not s.getLocation().getFile().getAbsolutePath().matches("%test%")
```

Comment:
That does not seem to catch global variables in a module. But perhaps I should try to extend `TestScope`, then?

Reply:
> That does not seem to catch global variables in a module. But perhaps I should try to extend `TestScope`, then?

:thinking: maybe this does fall far enough outside our normal test-detection that it's fine to do what you do. If there is an obvious way to extend our standard test-detection to handle this, that would be nice... otherwise I guess the point is that you might have `.../integration-tests/settings.py` and you don't want to show alerts about disabling CSRF protection from that `settings.py` file

Reply:
A module has a scope, so I could just apply the same heuristic as here to the module location, would that be acceptable?

Reply:
If you can make it fit nicely into the existing setup, or extend the existing setup with a new thing (such as `TestModule` or whatever), go for it

Reply:
I added a `TestModule` similarl to `UnknownTestFunction`.

Reply:
Good suggestion, thanks.